internal/server/server.go

```diff
@@ -18,7 +18,7 @@ func Server() {
 	r.HandleFunc("/v1/users/create", Serviceuser.CreateUser).Methods(http.MethodPost)
 	r.HandleFunc("/v1/users/login", Serviceuser.LoginUsers).Methods(http.MethodPost)
 	http.Handle("/v1/users/logout", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)))
-	http.Handle("/v1/users/{id}", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.GetCurrentProfile)))
+	r.HandleFunc("/v1/users/{id}", Serviceuser.GetCurrentProfile).Methods(http.MethodGet)
```

Comment:
why removing Auth middleware, if it's really "current profile" it has to have the authorized user